OneSignalSDK/unittest/src/test/java/com/test/onesignal/LocationIntegrationTests.java

```diff
@@ -0,0 +1,614 @@
+
+@Config(packageName = "com.onesignal.example",
+        shadows = {
+                ShadowOneSignalRestClient.class,
+                ShadowPushRegistratorFCM.class,
+                ShadowOSUtils.class,
+                ShadowAdvertisingIdProviderGPS.class,
+                ShadowCustomTabsClient.class,
+                ShadowCustomTabsSession.class,
+                ShadowHmsInstanceId.class,
+        },
+        sdk = 21
+)
+@RunWith(RobolectricTestRunner.class)
+public class LocationIntegrationTests {
+
+    private static final String ONESIGNAL_APP_ID = "b4f7f966-d8cc-11e4-bed1-df8f05be55ba";
+
+    @SuppressLint("StaticFieldLeak")
+    private static Activity blankActivity;
+    private static ActivityController<BlankActivity> blankActivityController;
+
+    private MockOSTimeImpl time;
+    private OSTrackerFactory trackerFactory;
+    private MockSessionManager sessionManager;
+    private MockOneSignalDBHelper dbHelper;
+
+    private static void cleanUp() throws Exception {
+        ShadowGMSLocationController.reset();
+
+        TestHelpers.beforeTestInitAndCleanup();
+
+        // Set remote_params GET response
+        setRemoteParamsGetHtmlResponse();
+    }
+
+    @BeforeClass // Runs only once, before any tests
+    public static void setUpClass() throws Exception {
+        ShadowLog.stream = System.out;
+
+        TestHelpers.beforeTestSuite();
+
+//        Field OneSignal_CurrentSubscription = OneSignal.class.getDeclaredField("subscribableStatus");
+//        OneSignal_CurrentSubscription.setAccessible(true);
+
+        OneSignal.setLogLevel(OneSignal.LOG_LEVEL.VERBOSE, OneSignal.LOG_LEVEL.NONE);
+        StaticResetHelper.saveStaticValues();
+    }
+
+    @Before
+    public void beforeEachTest() throws Exception {
+        blankActivityController = Robolectric.buildActivity(BlankActivity.class).create();
+        blankActivity = blankActivityController.get();
+
+        time = new MockOSTimeImpl();
+        trackerFactory = new OSTrackerFactory(new MockOSSharedPreferences(), new MockOSLog(), time);
+        sessionManager = new MockSessionManager(OneSignal_getSessionListener(), trackerFactory, new MockOSLog());
+        dbHelper = new MockOneSignalDBHelper(ApplicationProvider.getApplicationContext());
+
+        TestHelpers.setupTestWorkManager(blankActivity);
+
+        cleanUp();
+
+        OneSignal_setTime(time);
+    }
+
+    @After
+    public void afterEachTest() throws Exception {
+        afterTestCleanup();
+    }
+
+    @AfterClass
+    public static void afterEverything() throws Exception {
+        cleanUp();
+    }
+
+    @Test
+    @Config(shadows = {ShadowGoogleApiClientBuilder.class, ShadowGoogleApiClientCompatProxy.class, ShadowFusedLocationApiWrapper.class})
+    public void shouldUpdateAllLocationFieldsWhenTimeStampChanges() throws Exception {
+        ShadowApplication.getInstance().grantPermissions("android.permission.ACCESS_COARSE_LOCATION");
+        OneSignalInit();
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.